### PR TITLE
fixes #2296  After canceling a sharing, application goes back to the search menu

### LINKED ISSUE #2296: After canceling a sharing, application goes back to the search menu
**Summary:** 

On pressing back button after clicking on sharing button, application goes back to the search activity not to the image activity.

**Steps to reproduce:** 
->Menu 
->Explore
->Search something
->Click on an image
->Click on share button
->Press back button or scroll down to cancel sharing

**Device and Android version:** 

Pixel 2 XL emulator (Android 8.0) and Motorola G4 (Android 7.0)
 
**Commons app version:** 

Latest master.

**Would you like to work on the issue?**

Yes

### LINKED ISSUE #4063: After click on backButton, application goes back to the exploreActivity
**Summary:** 
when your are at SearchActivity  then click on Backbutton application goes back to exploreActivity .
I want application goes to RecentSearch fragment  and when click again on backbutton application goes back to exploreActivity.

**Steps to reproduce:** 
>Menu
->Explore
->Search something
->Click on BackButton

**Device and Android version:** 
API :29 
**Commons app version:** 
2.13.1
**Screen-shots:** 

**Would you like to work on the issue?**
YES
Please let us know whether you want to fix the issue by yourself. If not, anyone can get the issue assigned to them.
YES

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
**Description (required)**
Fixes #2296 

What changes did you make and why?
**remove the attribute -"addtoBackstack(null)"**
**Add conditions in OnBackPressed() of searchActivity to manage fragments when back button is pressed**

**Tests performed (required)**
Android Device :Redmi Y3
API :29

**Screenshots (for UI changes only)**

### CODE REVIEW

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul review my this pull request

**Comment by Prince-kushwaha** [comment]
@misaochan review my PR

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul  review my PR

**Comment by Prince-kushwaha** [comment]
@misaochan review my PR

File: app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java [hunk 1]
```diff
@@ -211,49 +211,32 @@ public void onMediaClicked(int index) {
...
         mediaDetails.showImage(index);
     }
 
-    /**
-     * This method is called on Screen Rotation
-     */
-    @Override
-    protected void onResume() {
-        if (supportFragmentManager.getBackStackEntryCount()==1){
-            //FIXME: Temporary fix for screen rotation inside media details. If we don't call onBackPressed then fragment stack is increasing every time.
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java:230]: This code seems like a temporary fix for orientation change, why did you remove this one?

File: app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java [hunk 2]
```diff
@@ -211,49 +211,32 @@ public void onMediaClicked(int index) {
                     .beginTransaction()
                     .hide(supportFragmentManager.getFragments().get(supportFragmentManager.getBackStackEntryCount()))
                     .add(R.id.mediaContainer, mediaDetails)
-                    .addToBackStack(null)
                     .commit();
-            // Reason for using hide, add instead of replace is to maintain scroll position after
+            // Reason for using hide, add instead of replace is to maintain scroll0 position after
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java:215]: Is scroll0 a typo?

**Review by neslihanturan — CHANGES_REQUESTED** [review]
Thanks for the PR @Prince-kushwaha , please see my review.

File: app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java (hunk 2 above)
Comment (Prince-kushwaha) [reply app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java:215]: yes i fix this

File: app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java (hunk 1 above)
Comment (Prince-kushwaha) [reply app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java:230]: we check above  condition when we  added fragment in   supportFragementManager backstack
but we are not add any fragement in fragementManager backstack
To add fragement in fragementManager backstack we use `supportFragementManager.addToBackStack(null)
`

`

Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java:230]: Sorry, I couldn't understand your explanation. Can you explain it again?
